src/vs/base/browser/ui/grid/grid.ts

```diff
@@ -217,9 +217,17 @@ export class Grid<T extends IView = IView> extends Disposable {
 
 	private didLayout = false;
 
-	constructor(view: T, options: IGridOptions = {}) {
+	constructor(gridview: GridView, options?: IGridOptions);
+	constructor(view: T, options?: IGridOptions);
+	constructor(view: T | GridView, options: IGridOptions = {}) {
 		super();
-		this.gridview = new GridView(options);
+
+		if (view instanceof GridView) {
```

Comment:
Can you please clarify this new case when the view is instanceof GridView. Since this seems to not be covered before

Reply:
Grid historically created a grid view and then added views, we know this has pitfalls due to layouts for each added view. This allows us to provide a preconstructed one.

Reply:
Ya this is weird, I'd rather have a `public deserialize` method for this. But that's just API preference.